store/store.go

```diff
@@ -1104,6 +1147,42 @@ func (s *Store) Sections(user *auth.UserState) ([]string, error) {
 	return sectionNames, nil
 }
 
+// WriteCommandsCatalogs queries the "commands" endpoint and writes
+// the command names into the given io.Writer [and at some point the
+// CNF catalog to another io.Writer; adding now to minimise interface
+// churn].
+func (s *Store) WriteCommandsCatalogs(names io.Writer, _ io.Writer) error {
+	u := *s.commandsURI
+
+	q := u.Query()
+	if release.OnClassic {
+		q.Set("confinement", "strict,classic")
+	} else {
+		q.Set("confinement", "strict")
+	}
+
+	u.RawQuery = q.Encode()
+	reqOptions := &requestOptions{
+		Method: "GET",
+		URL:    &u,
+		Accept: halJsonContentType,
+	}
+
+	resp, err := httputil.RetryRequest(u.String(), func() (*http.Response, error) {
+		return s.doRequest(context.TODO(), s.client, reqOptions, nil)
+	}, func(resp *http.Response) error {
+		return decodeCatalog(resp, names)
+	}, defaultRetryStrategy)
```

Comment:
Quite hard to follow what's going on here. This will probably improve much if we define the arguments ahead of the function call and then have a single line with the call.

Reply:
agreed! Any objections to shuffling the other uses of retry to be similar, in the same pr?

Reply:
No objections since the PR is about to go in. It's usually bad to mix these changes in when it's still pending reviews, since the reviewer would then spend 5 minutes just looking at those lines to figure that there were no changes. :)